shared/packages/api/src/inputApi.ts

```diff
@@ -264,10 +326,16 @@ export namespace Accessor {
 		allowWrite: false
 
 		/** Base url (url to the host), for example http://myhost.com/fileShare/ */
-		baseUrl: string
+		baseUrl?: string
 
 		/** Name/Id of the network the share exists on. Used to differ between different local networks. Leave empty if globally accessible. */
 		networkId?: string
+
+		/** If true, assumes that a source never changes once it has been fetched. */
+		isImmutable?: boolean
+
+		/** If true, assumes that the source supports HEAD requests. Otherwise, GET requests will be sent to check availability. */
+		supportHEAD?: boolean
```

Comment:
I think we should _assume_ that HEAD is available, and only use GET as an exception.

```suggestion
		/** If true, assumes that the source supports HEAD requests. Otherwise, GET requests will be sent to check availability. */
		useGETinsteadOfHEAD?: boolean
```